extensions/common/json-ld/src/main/java/org/eclipse/edc/jsonld/transformer/TransformerUtil.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.jsonld.transformer;
+
+import jakarta.json.JsonArray;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonString;
+import jakarta.json.JsonValue;
+import org.eclipse.edc.transform.spi.TransformerContext;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static org.eclipse.edc.jsonld.transformer.JsonLdKeywords.VALUE;
+
+public class TransformerUtil {
+    
+    private TransformerUtil() { }
+    
+    public static Object transformGenericProperty(JsonValue value, TransformerContext context) {
+        if (value instanceof JsonArray) {
+            var jsonArray = (JsonArray) value;
+            if (jsonArray.isEmpty()) {
```

Comment:
this first condition is not needed, if the size is 0 it could finish in the `else` branch, the result will be the same.